content/en/integrations/kubernetes_state_core.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Kubernetes State Metrics Core
+name: kubernetes_state_core
+kind: integration
+description: "Monitor the health of the workload running on your Kubernetes cluster. Track the status of your Kubernetes objects, correlate your microservice metrics, and more."
+short_description: "Track the status of your Kubernetes objects, correlate your microservice metrics, and more."
+categories:
+    - cloud
+    - configuration & deployment
+    - containers
+    - orchestration
+doc_link: /integrations/kubernetes_state_core/
+dependencies:
+    ['https://github.com/DataDog/documentation/blob/master/content/en/integrations/kubernetes_state_core.md']
+has_logo: true
+integration_title: Kubernetes State Metrics Core
+is_public: true
+public_title: Datadog-Kubernetes State Metrics Core Integration
+integration_id: "kubernetes_state_core"
+---
+
+## Overview
+
+Get metrics from Kubernetes service in real time to:
+
+- Visualize and monitor kubernetes states
+- Be notified about kubernetes failovers and events.
+
+## Setup
+
+### Installation
+
+The Kubernetes State Metrics Core check is included in the [Datadog Cluster Agent][1] image, so you don't need to install anything else on your Kubernetes servers.
+
+### Requirements
+
+- Datadog Cluster Agent v1.12+ 
+
+### Configuration
+
+We highly recommend using the Datadog Helm Chart to configure the Kubernetes State Metrics Core check.
+
+In your Helm `values.yaml`, add the following:
+
+```
+...
+datadog:
+...
+  kubeStateMetricsCore:
+    enabled: true
+...
+```
+
+**Note:** Enabling `kubeStateMetricsCore` in your Helm `values.yaml` will also configure the Agent to ignore the auto configuration file for legacy `kubernetes_state` check. The goal is to avoid running both checks simultaneously.
+
+**Optional:** The Kubernetes State Metrics Core check does not require deploying `kube-state-metrics` in your cluster, you can optionally disable deploying `kube-state-metrics` as part of the Datadog Helm Chart. To do this, add the following in your Helm `values.yaml`:
+
+```
+...
+datadog:
+...
+  kubeStateMetricsEnabled: false
```

Comment:
shouldn't we do the opposite? not deploy KSM when the core check is enabled, unless the user enables this?

Reply:
`kubeStateMetricsEnabled ` is enabled by default actually

Reply:
Yes i think I didn't explain myself fully. The current defaults are good to me as well. To paraphrase, I think that we should make it more explicit on what happens if kubeStateMetricsCore is enabled:
1) Disabling the legacy `kubernetes_state` and ignoring it's auto configuration file (as well as the custom config if exists?)
2) `kube-state-metrics` will not be deployed via our chart?

Reply:
> Disabling the legacy kubernetes_state and ignoring it's auto configuration file (as well as the custom config if exists?)

yes, this is documented in the note just above. 

> kube-state-metrics will not be deployed via our chart?

This is not what I think we should do, here is why:
Since deploying ksm is enabled by default, we can't really judge whether it's enabled on purpose or not, so disabling it just because the core check is enabled is a tricky decision. A user can be interested in enabling both the core check and the python check at the same time during the migration phase, so forcing disabling the ksm deployment can be undesired. what we can do though is to print a helm note to tell the user they can (or should) disable `kubeStateMetricsEnabled` if `kubeStateMetricsCore` is enabled.

Nothing is decided yet though, happy to hear feedback on this!

Reply:
This definitely makes sense to me. Thanks for sharing your thoughts!